cloudinit/sources/DataSourceAzure.py

```diff
@@ -660,10 +691,249 @@ def setup(self, is_new_instance):
             LOG.debug("negotiating already done for %s",
                       self.get_instance_id())
 
+    def _wait_for_nic_detach(self, nl_sock):
```

Comment:
We decorate `_wait_for_hot_attached_nics` with `@azure_ds_telemetry_reporter`. Decorate this one for telemetry logging and consistency.

```suggestion
    @azure_ds_telemetry_reporter
    def _wait_for_nic_detach(self, nl_sock):
```